### PR TITLE
112 improve exception handling

### LINKED ISSUE #112: Improve exception handling
We want to inform users when an error occurs during a request.

The backend needs to handle all errors and return them clearly for the frontend to use.

Acceptance Criteria :
- [x] every route handles unknown exceptions
- [x] each route handles custom exceptions that clearly indicate what has happened

### PR DESCRIPTION
**Pull request by sylvanie85**
ref #112

### CODE REVIEW

**Comment by sylvanie85** [comment]
Hi @Francois-Werbrouck and @ChromaticPanic here's a PR to improve exception handling on the backend.  What do you think needs to be added? 

I think I'll also create the same issue for the datastore to check if all nachet functions and requests handle relevant exceptions.

**Review by ChromaticPanic — APPROVED** [review]

File: app.py [hunk 1]
```diff
@@ -190,14 +187,23 @@ async def get_user_id() :
...
         user_id = datastore.get_user_id(email)
         
-        return jsonify({"user_id": user_id}), 200    
-    except (KeyError, TypeError, ValueError, datastore.DatastoreError) as error:
+        return jsonify({"user_id": user_id}), 200 
+    
+    except datastore.DatastoreError as error:
         print(error)
-        return jsonify([f"GetUserIdError: {str(error)}"]), 400
+        return jsonify([f"Datastore Error retrieving user id for email {email} : {str(error)}"]), 400
```
Comment (Francois-Werbrouck) [inline app.py:200]: Should we make this a different return code?

**Review by Francois-Werbrouck — APPROVED** [review]
LGTM

File: app.py (hunk 1 above)
Comment (sylvanie85) [reply app.py:200]: I don't know if it's relevant to use different codes, it depends on the frontend I think, what do you think @ChromaticPanic?

**Pull request merged**